### PR TITLE
[0-size Tensor Job2 No.26、62、103、104] Add 0-size Tensor support for paddle.incubate.nn.functional.fused_bias_dropout_residual_layer_norm [fluid_ops]

### PR DESCRIPTION
**Pull request by co63oc**
### PR Category

Operator Mechanism

### PR Types

Bug fixes

### Description

26 paddle.incubate.nn.functional.fused_bias_dropout_residual_layer_norm
https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/incubate/nn/functional/fused_bias_dropout_residual_layer_norm_cn.html#fused-bias-dropout-residual-layer-norm
image

输出与x相同，但是测试residual如果为0-size，和residual相同，修改infermeta，没有符号推导
反向填充0，PaddeAPITest为自定义规则
https://github.com/PFCCLab/PaddleAPITest/blob/ec3cd8547f76e608c492f8a7ddb941c64765d378/tester/paddle_to_torch/rules.py#L2256

PaddleAPITest测试通过
image

62 paddle.nn.functional.margin_cross_entropy
https://github.com/PFCCLab/PaddleAPITest/pull/354

margin_cross_entropy 增加logits结尾维度不为0的检查
image

103 deform_conv2d
文档
https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/vision/ops/deform_conv2d_cn.html
输入输出说明
image
从文档查看，输入0-size，输出可能不是0-size，测试中有不同规则检查，没有找到对应测试用例，代码还是按填充编写

修改前向和反向，使用0填充
torch测试反向填充
```
import torch
import torchvision
_kwargs = {}
_kwargs["input"] = torch.randn([0, 2, 5, 5])
_kwargs["input"].requires_grad = True
_kwargs["offset"] = torch.randn([0, 18, 3, 3])
_kwargs["weight"] = torch.randn([1, 2, 3, 3])
_kwargs["weight"].requires_grad = True
_kwargs["bias"] = None
_kwargs["stride"] = [1, 1]
_kwargs["padding"] = [0, 0]
_kwargs["dilation"] = [1, 1]

result = torchvision.ops.deform_conv2d(**_kwargs)
result.sum().backward()
print(_kwargs["weight"].grad)

tensor([[[[0., 0., 0.],
          [0., 0., 0.],
          [0., 0., 0.]],

         [[0., 0., 0.],
          [0., 0., 0.],
          [0., 0., 0.]]]])
```

PaddleAPITest测试通过
image

错误中断是torch错误
image

104 distribute_fpn_proposals 
增加单测

文档地址 https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/vision/ops/distribute_fpn_proposals_cn.html
image

PaddleAPITest 测试通过，错误为numpy error
image

### CODE REVIEW

**Comment by DanielSun11** [comment]
请解决下conflicts

**Comment by co63oc** [comment]
@DanielSun11 CI已完成需要review

**Review by DanielSun11 — APPROVED** [review]
LGTM
